cli/internal/runsummary/spaces.go

```diff
@@ -94,3 +102,34 @@ func newSpacesTaskPayload(taskSummary *TaskSummary) *spacesTask {
 		Logs:         string(taskSummary.GetLogs()),
 	}
 }
+
+// getStateOfRepo returns the sha and branch when in a git repo
+// Otherwise it should return empty strings right now.
+// We my add handling of other scms and non-git tracking in the future.
+func getStateOfRepo() (string, string) {
```

Comment:
I really wish Rust had named return types. Returning tuples is such an easy way to introduce a bug

Reply:
hah well `getStateOfRepo` certainly conflicts with `RepoState` (which is now ExecutionState?) from rust land :)

I didn't want to say `git`, because I wanted to convey that we want this to handle non-git and other vcs scenarios also. But maybe that's YAGNI, i'll take your suggestion